internal/zinc-persist/src/main/scala/sbt/internal/inc/RelationsTextFormat.scala

```diff
@@ -0,0 +1,128 @@
+package sbt.internal.inc
+
+import java.io.{ File, BufferedReader, Writer }
+
+import sbt.internal.util.Relation
+import xsbti.api.DependencyContext._
+
+trait RelationsTextFormat extends FormatCommons {
+
+  def mappers: AnalysisMappers
+
+  private case class Descriptor[A, B](
+    header: String,
+    selectCorresponding: Relations => Relation[A, B],
+    keyMapper: Mapper[A],
+    valueMapper: Mapper[B]
+  )
+
+  private def stringsDescriptor(header: String, rels: Relations => Relation[String, String]) =
+    Descriptor(header, rels, Mapper.forString, Mapper.forString)
+
+  private val allRelations: List[Descriptor[_, _]] = {
```

Comment:
This should probably be well documented and linked with the original source `Relations` since any change there may affect behaviour here. I would do the same for the rest of the methods in this class. In fact, it's a little bit unfortunate that this is generated manually and not automatically.